lib/lroPollStrategy.ts

```diff
@@ -290,11 +265,50 @@ class LocationLROPollStrategy extends LROPollStrategy {
         (statusCode === 204 && (this._initialRequestMethod === "DELETE" || this._initialRequestMethod === "POST"))) {
         this._status = "Succeeded";
         this._resource = getResponseBody(result);
+      } else if (statusCode === 404 && this._initialRequestMethod === "POST" &&
+        (this._initialRequestStatusCode === 200 || this._initialRequestStatusCode === 201 || this._initialRequestStatusCode === 202)) {
+        this._status = "Failed";
+        this._resource = getResponseBody(result);
       } else {
         throw new Error(`The response with status code ${statusCode} from polling for long running operation url "${this._locationHeaderValue}" is not valid.`);
       }
     });
   }
+
+  protected finalStatusIsAcceptable(): boolean {
+    return this._status === "Succeeded" ||
+      (this._initialRequestMethod === "POST" && this._response.status === 404 &&
+        (this._initialRequestStatusCode === 200 ||
```

Comment:
The name initialRequestStatusCode smells funny to me because requests don't have status codes.

Reply:
Also, I'd like if repeated checks on instance vars like this could be extracted to a local variable whose name can be mangled.

Reply:
Variable renamed in https://github.com/Azure/ms-rest-azure-js/pull/43/commits/8920e1512c423b5efc668fa50715915f7ddc48b9.